google-cloud-spanner/src/test/java/com/google/cloud/spanner/ReadWriteTransactionWithInlineBeginTest.java

```diff
@@ -389,6 +404,73 @@ public void failedBatchUpdateAndThenUpdate() {
     assertThat(countTransactionsStarted()).isEqualTo(2);
   }
 
+  @Test
+  public void executeSqlWithOptimisticConcurrencyControl() {
+    client
+        .readWriteTransaction(Options.optimisticLock())
+        .run(
+            transaction -> {
+              try (ResultSet rs = transaction.executeQuery(SELECT1)) {
+                while (rs.next()) {
+                  assertEquals(rs.getLong(0), 1);
+                }
+              }
+              return null;
+            });
+    Collection<AbstractMessage> requests =
+        Collections2.filter(
+            mockSpanner.getRequests(), msg -> msg.getClass().equals(ExecuteSqlRequest.class));
+    assertEquals(requests.size(), 1);
+    ExecuteSqlRequest request = (ExecuteSqlRequest) Iterables.getOnlyElement(requests);
```

Comment:
Can we avoid import an additional library Collections2 here, for filtering? 

We can use [streams and lambdas](https://stackoverflow.com/a/1385698) to achieve the same.

Reply:
sure. done.